lib/wibox/widget/slider.lua

```diff
@@ -0,0 +1,473 @@
+        or beautiful.slider_handle_margins
+
+    x_offset, y_offset = 0, 0
+
+    if margins then
+        if type(margins) == "number" then
+            x_offset, y_offset = margins, margins
+            handle_width  = handle_width  - 2*margins
+            handle_height = handle_height - 2*margins
+        else
+            x_offset, y_offset = margins.left or 0, margins.top or 0
+            handle_width  = handle_width  -
+                (margins.left or 0) - (margins.right  or 0)
+            handle_height = handle_height -
+                (margins.top  or 0) - (margins.bottom or 0)
+        end
+    end
+
+    local value = self._private.value or self._private.min or 0
+
+    -- Get the widget size back to it's non-transfored value
+    local min, _, interval = get_extremums(self)
+    local rel_value = ((value-min)/interval) * (width-handle_width)
+
+    cr:translate(x_offset + rel_value, y_offset)
+
+    local handle_border_width = self._private.handle_border_width
+        or beautiful.slider_handle_border_width
+        or properties.handle_border_width or 0
+
+    handle_shape(cr, handle_width, handle_height)
+
+    if handle_border_width > 0 then
+        cr:fill_preserve()
+    else
+        cr:fill()
+    end
+
+    -- Draw the handle border
+    if handle_border_width > 0 then
+        local handle_border_color = self._private.handle_border_color
+            or beautiful.slider_handle_border_color
+            or properties.handle_border_color
+
+        if handle_border_color then
+            cr:set_source(color(handle_border_color))
+        end
+
+        cr:set_line_width(handle_border_width)
+        cr:stroke()
+    end
+end
+
+function slider:fit(_, width, height)
+    -- Use all the space, this should be used with a constraint widget
+    return width, height
+end
+
+local event = {}
+
+-- Move the handle to the correct location
+local function move_handle(self, x, _, width)
+    self:set_value(get_value(self, width, x))
+end
+
+function event.mouse_press(self, x, y, button_id, _, geo)
+    if button_id ~= 1 then return end
+
+    local matrix_from_device = geo.hierarchy:get_matrix_to_device():invert()
+
+    -- Sigh. geo.width/geo.height is in device space. We need it in our own
+    -- coordinate system
+    local _, _, width, _ = matrix_from_device:transform_rectangle(geo.x, geo.y, geo.width, geo.height)
+
+    move_handle(self, x, y, width)
+
+    -- Calculate a matrix transforming from screen coordinates into widget coordinates
+    local wgeo = geo.drawable.drawable:geometry()
+    local matrix = matrix_from_device:translate(-wgeo.x, -wgeo.y)
+
+    capi.mousegrabber.run(function(mouse)
+        if not mouse.buttons[1] then
+            self:emit_signal("widget::redraw_needed")
+            return false
+        end
+
+        -- Calculate the point relative to the widget
+        local x_prime, y_prime = matrix:transform_point(mouse.x, mouse.y)
+        move_handle(self, x_prime, y_prime, width)
```

Comment:
I'm exaggerating, but: Change `move_handle()`'s argument order into `self, width, x, y` and the above can become `move_handle(self, width, matrix:transform_point(mouse.x, mouse.y))`. (Feel free to ignore this comment, I was only wondering about the variable names)